### PR TITLE
show legend for waveform plots

### PR DESCRIPTION
**Pull request by Bechir-Brahem**
fix for issue #131 
show legend for waveform plots. did not add any new code just moved lines around.

in my opinion I find the legend right now more annoying than useful.

merged with PR #136

### CODE REVIEW

File: pyctbgui/services/ADC.py [hunk 1]
```diff
@@ -26,6 +26,7 @@ def setup_ui(self):
 
     def initializeAllAnalogPlots(self):
         self.mainWindow.plotAnalogWaveform = pg.plot()
+        self.mainWindow.plotAnalogWaveform.addLegend(colCount=4)
```
Comment (erikfrojdh) [inline pyctbgui/services/ADC.py:29]: colCount=4 appears 4 times is this a coincidence or should it be a named variable?

**Review by erikfrojdh — COMMENTED** [review]
See comment, if just by coincidence I'll merge.

**Comment by Bechir-Brahem** [comment]
added colCount in Defines

**Comment by erikfrojdh** [comment]
Looking at your demo I think the legend is more in the way than helping. We anyway can see the colors from the settings on the left. @thattil: What is your opinion on this?

**Comment by thattil** [comment]
@erikfrojdh @Bechir-Brahem I had a look at the PR and I see that all the legends show despite if they are selected or not. This is a bug from my side. Initially, only the selected waveforms and legends showed. I changed this to adding them all in the beginning and showing/hiding the selected/unselected waveforms, but I missed about the legends. 

If only the selected legends were shown, then maybe its not a bad idea to show them. Also I doubt all of them are selected everytime. By default, the number of columns for the legends is chosen as 4, but they could also be selected in a more intelligent manner depending on the number of selected waveforms. This is an idea that I can ask the users directly first to see if they prefer this or hiding all the legends.

**Review by thattil — CHANGES_REQUESTED** [review]
I would show selected legend by default.. and hide only if user sets it..

**Review by thattil — APPROVED** [review]

**Pull request merged**